enrolkey_form.php

```diff
@@ -0,0 +1,32 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+//moodleform is defined in formslib.php
+require_once($CFG->libdir . '/formslib.php');
+
+/**
+ * Class enrolkey_form
+ */
+class enrolkey_form extends moodleform {
+    
+    //Add elements to form
+    public function definition() {
+        $mform = $this->_form;
+        $mform->addElement('text', 'enrolkey', '');
+        $mform->setType('enrolkey', PARAM_TEXT);
+        $mform->addElement('submit', 'submitbutton', get_string('enrolbutton', 'block_enrolkey'));
+    }
```

Comment:
I would move some of the validation logic into here.

* [ ] ie if the key is empty then show an error (set the element as require in definition()
* [ ] if the key doesn't match any keys then say it in invalid rather than redirecting to /my/